src/components/Flyout/Flyout.tsx

```diff
@@ -0,0 +1,47 @@
+import { useRef } from 'react';
+import { useSelectionStore } from '../../store/useSelectionStore';
+import { downloadItemsAsCSV } from '../../utils/downloadItemsAsCSV';
+
+export default function Flyout() {
+  const { getSelectedArray, getSelectedCount, unselectAll } =
+    useSelectionStore();
+
+  const count = getSelectedCount();
+  const items = getSelectedArray();
+  const linkRef = useRef<HTMLAnchorElement>(null);
```

Comment:
Does this component work as you expected?
Don't you think that `linkRef` is always equal `null` and you'll never enter inside the `if` statement in the `handleDownload`